transformer_engine/pytorch/float8_tensor.py

```diff
@@ -457,21 +457,21 @@ def transpose(
                 )
             return super().transpose(dim0, dim1)
 
+        if not cache:
+            # Use optimized kernel for basic 2D transpose
+            # TODO Support differentiation # pylint: disable=fixme
+            return Float8Tensor.make_like(
+                self,
+                data=tex.fp8_transpose(
+                    self._data.contiguous().detach(),
+                    self._fp8_dtype,
+                ),
+            )
+
         # Handle caching
-        if cache and self._transpose is None:
-            self._transpose = self.transpose(cache=False)
-        if self._transpose is not None:
-            return self._transpose
-
-        # Use optimized kernel for basic 2D transpose
-        # TODO Support differentiation # pylint: disable=fixme
-        return Float8Tensor.make_like(
-            self,
-            data=tex.fp8_transpose(
-                self._data.contiguous().detach(),
-                self._fp8_dtype,
-            ),
-        )
+        if cache or self._transpose is None:
```

Comment:
`cache == True` at this point:
```suggestion
        if self._transpose is None:
```